AndroidSDKCore/src/main/java/com/leanplum/messagetemplates/BaseMessageDialog.java

```diff
@@ -129,17 +132,55 @@ protected BaseMessageDialog(Activity activity, boolean fullscreen, BaseMessageOp
           window.setDimAmount(0.7f);
         }
       } else {
+        if (htmlOptions == null) {
+          return;
+        }
+
         window.clearFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND);
-        window.setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL,
-            WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL);
-        if (htmlOptions != null &&
-            MessageTemplates.Args.HTML_ALIGN_BOTTOM.equals(htmlOptions.getHtmlAlign())) {
-          dialogView.setGravity(Gravity.BOTTOM);
+        boolean treatAsBanner = treatAsBanner(htmlOptions);
+        if (treatAsBanner) {
+          // banners need to be positioned at the top manually (unless they get repositioned to the bottom later)
+          // or the anchor would be the center of the screen
+          window.setLayout(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
+          window.setGravity(Gravity.TOP);
+
+          // use the html y offset to determine the y location of the window; this is different
+          // from non-banners because we don't want to make the window too big (e.g. via a margin
+          // in the layout) and block other things on the screen (e.g. dialogs)
+          WindowManager.LayoutParams windowLayoutParams = window.getAttributes();
+          windowLayoutParams.y = htmlOptions.getHtmlYOffset(activity);
+          window.setAttributes(windowLayoutParams);
+
+          window.setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL | WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE,
+              WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL | WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE);
+        } else {
+          window.setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL,
+              WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL);
+        }
+
+        if (MessageTemplates.Args.HTML_ALIGN_BOTTOM.equals(htmlOptions.getHtmlAlign())) {
+          if (treatAsBanner) {
+            window.setGravity(Gravity.BOTTOM);
+          } else {
+            dialogView.setGravity(Gravity.BOTTOM);
+          }
         }
       }
     }
   }
 
+  /**
+   * Banners with property TabOutsideToClose = false need to be treated differently
```

Comment:
did you mean TapOutside instead of TabOutside?

Reply:
yes, but the existing function and vars are named is `isHtmlTabOutsideToClose()`, i just left them they are...